Created rspatialdata blogpost (#247)
ropensci/roweb3: content/blog/2021-09-14-rspatialdata/index.Rmd

date: '2021-09-14'
slug: rspatialdata
categories: []
tags:
  - rnaturalearth
  - osmdata
  - MODIStsp
  - nasapower
  - rdhs
  - open-data
  - spatial
  - geospatial
  - tutorial
  - community
description: "rspatialdata: tutorials for working with spatial data using R, featuring many rOpenSci packages!"
twitterImg: blog/2021/09/14/rspatialdata/rspatialdata-home.jpg
twitterAlt: "Screenshot of rspatialdata homepage featuring multicoloured tiles each outlining a different type of spatial data"
output:
  html_document:
    keep_md: yes
---

As two postgrad students on summer vacation but with no travel plans (during this global pandemic of course), we took up an internship at King Abdullah University of Science and Technology (KAUST) last summer and ended up collaborating on a really cool project with [Paula Moraga](https://www.paulamoraga.com/)! Did we mention that we worked in Saudi Arabia, while living in Australia? All is well when we have the internet.

We also ended up [presenting our project at useR! 2021](https://www.youtube.com/watch?v=nLikORFBKgk), and winning an award for the most outstanding lightning talk! So, keep reading to see how we got there!

> Although there is so much data being collected in multiple disciplines and made openly available, not many know how to retrieve and utilize these resources. Spatial and spatio-temporal data are two of the most commonly collected data types and also used in research across a wide range of domains.

Reviewer: Can "not many know how to ..." please be changed to something that does not imply inability on the part of people, like, "it may be difficult to know ...", or "it can be difficult to find out how to ..." or something like that?

Reviewer: I'm also a little unsure about the claim that
> Spatial and spatio-temporal data are two of the most commonly collected data types 

but i'm equally unsure about how that might be improved? No need to change that unless you can see a better way.

> With the R programming language becoming increasingly popular among academics, researchers, and scientists, and because spatial data is easier to interpret when visualized, Paula thought it would be a great idea to create a repository of data sources and simple tutorials on how to retrieve and visualize spatial data using R.

Reviewer: Suggestion, with request for clarification from @Paula-Moraga:

> ~~Paula~~ We though it would be a great idea ...

Otherwise the reader starts off thinking this was entirely Paula's idea, when I'd suspect a lot of ideas came from @dilinieseimon and Varsha as well.

Reviewer 2: Maybe this: Paula thought that a great project for a KAUST internship would be to create a repository

> This idea then became [rspatialdata](https://rspatialdata.github.io/) - a collection of data sources and tutorials on downloading and visualising spatial data using R.

Reviewer: > This idea ~~then~~ became

{{< figure src = "rspatialdata-home.jpg" width = "800" alt = "Screenshot of rspatialdata homepage featuring multicoloured tiles each outlining a different type of spatial data" class = "center" link = "https://rspatialdata.github.io/">}}

> rspatialdata consists of a non-exhaustive list of R-packages which have been developed as clients for different spatio-temporal databases and APIs. It also consists of tutorials on how to use these R packages, understand the different types of spatial data objects available and create visualizations using them. After doing a lot of research, we managed to pick out a few R-packages which we were confident gave up-to-date data from reliable sources.

Reviewer: Please hyperlink first `rspatialdata`

Coincidentally, much of what we picked out happened to be rOpenSci packages! We were amazed at how rich and easy to use most of these packages were. So here is our experience using those packages and some things we really enjoyed doing!

## Downloading boundaries of countries using [rnaturalearth](https://docs.ropensci.org/rnaturalearth)

> The rnaturalearth package (by [Andy South](https://github.com/andysouth)) facilitates communication with [Natural earth](http://www.naturalearthdata.com/) map data. It allows us to easily download and visualize boundaries of countries and boundaries of states within countries as well.

Reviewer: I'm not sure here, but think maybe changing
> It allows us to easily download ...

to
> It allows you to easily download ...

makes it seem more inclusive and engaging? What do you think @steffilazerte?

We used rnaturalearth to download the boundaries of countries in many of our tutorials - including our tutorials on [elevation](https://rspatialdata.github.io/elevation.html), [rainfall](https://rspatialdata.github.io/rainfall.html) and [humidity](https://rspatialdata.github.io/humidity.html).

## Visualizing hospitals in Lagos, Nigeria using [osmdata](https://docs.ropensci.org/osmdata/)

> osmdata (by [Mark Padgham](/author/mark-padgham/)) is an R package for downloading spatial data from [OpenStreetMap (OSM)](https://www.openstreetmap.org/) - a very cool open source project. We were amazed at the huge variety of spatial features available to us! We were able to download spatial data about almost anything from amenities such as colleges, cinemas, hospitals and banks, to different types of highways and streets such as motorways, residential streets and service lanes.

Reviewer: The package actually has 4 authors, all of whom should be mentioned here: @robinlovelace, @maelle, and @hrbrmstr as well. Maybe check the other authorial attributions as well, to ensure all authors are appropriately listed?

Reviewer: Plus a personal political request, but could the final list maybe be changed so "motorways" does not come first? A big motivation for my software is to help move away from car-centric views of the world. "walking and bicycle paths" instead of "motorways" maybe?

Reviewer 3: Thanks guys, this looks set to be a great blog post, thanks for the link-up @mpadge, and the article @dilinieseimon !

Reviewer 4: That was my bad! I added the authors in and didn't catch that!

We used the osmdata package to download and visualize hospitals in Lagos, Nigeria. Then we also downloaded spatial data for different types of highways, streets and waterways in Lagos, and created the following map entirely using data retrieved through osmdata!

{{< figure src = "hospitals-lagos-nigeria.jpg" width = "600" alt = "A black and white map of Logos, Nigeria, showing highways in orange, smaller streets in grey, rivers in light blue, and navy blue shapes outlining hospital locations" class = "center" link = "https://rspatialdata.github.io/osm.html">}}

Have a read through our [Open Street Map data tutorial](https://rspatialdata.github.io/osm.html) for a complete tutorial on how to install the osmdata package, find what spatial features are available, learn how to download these features and also create some cool maps (including an interactive one)!

## Visualizing vegetation in Mongolia and land cover in Zimbabwe using [MODIStsp](https://docs.ropensci.org/MODIStsp/)

> [MODIS (Moderate Resolution Imaging Spectroradiometer)](https://modis.gsfc.nasa.gov/) is an instrument aboard the NASA Terra and Aqua satellites, which orbits the entire Earth every 1-2 days, acquiring data at different spatial resolutions. The data acquired by MODIS describes features of the land, oceans and the atmosphere.

Reviewer: > Terra and Aqua satellites, which orbit~~s~~ the entire Earth

We also used the [MODIS Land Cover Type Products](https://modis.gsfc.nasa.gov/data/dataprod/mod12.php) to visualize the land cover classification in Zimbabwe.

{{< figure src = "land-cover-zimbabwe.jpg" width = "600" alt = "A map of Zimbabwe with its area colored by the corresponding land cover type - such as grasslands, croplands, barren etc." class = "center" link = "https://rspatialdata.github.io/land_cover.html">}}

Using the MODIStsp package made it so much easier to find what data products were available and also directly download and save this data using R. Our [Vegetation](https://rspatialdata.github.io/vegetation.html) and [Land Cover](https://rspatialdata.github.io/land_cover.html) tutorials explain how we used the MODIStsp R package to download data and create cool maps.

## Visualizing rainfall in Gansu, China and relative humidity in Western Australia using [nasapower](https://docs.ropensci.org/nasapower/)

> nasapower (by [Adam Sparks](/author/adam-sparks/)) is a client for the '[NASA POWER](https://power.larc.nasa.gov/)' global meteorology, surface solar energy and climatology data API, and aims to make it quick and easy to download climatology data for analysis, visualization, modeling and many other purposes.

Reviewer: Also please hyperlink this first `nasapower` reference here

Reviewer 4: It's hyperlinked in the title, do you think it should also be hyperlinked in the text too?

We used the nasapower package to retrieve rainfall and humidity data, and it was such a simple task. We only had to submit the duration of data we needed along with the geographical location, and the package took care of all the hard work.

> We downloaded relative humidity and rainfall data using the nasapower package and then visualized the relative humidity in Western Australia and the rainfall in Gansu, China.

Reviewer: > We downloaded relative humidity and rainfall data ~~using the nasapower package~~ and then visualized

- otherwise a bit repetitive, and it is plenty clear enough at that stage that the nasapower package was used.

date: '2021-09-14'
slug: rspatialdata
categories: []
tags:
  - rnaturalearth
  - osmdata
  - MODIStsp
  - nasapower
  - rdhs
  - open-data
  - spatial
  - geospatial
  - tutorial
  - community
description: "rspatialdata: tutorials for working with spatial data using R, featuring many rOpenSci packages!"
twitterImg: blog/2021/09/14/rspatialdata/rspatialdata-home.jpg
twitterAlt: "Screenshot of rspatialdata homepage featuring multicoloured tiles each outlining a different type of spatial data"
output:
  html_document:
    keep_md: yes
---

As two postgrad students on summer vacation but with no travel plans (during this global pandemic of course), we took up an internship at King Abdullah University of Science and Technology (KAUST) last summer and ended up collaborating on a really cool project with [Paula Moraga](https://www.paulamoraga.com/)! Did we mention that we worked in Saudi Arabia, while living in Australia? All is well when we have the internet.

We also ended up [presenting our project at useR! 2021](https://www.youtube.com/watch?v=nLikORFBKgk), and winning an award for the most outstanding lightning talk! So, keep reading to see how we got there!

> Although there is so much data being collected in multiple disciplines and made openly available, it may be difficult to find, retrieve and utilize these resources. Spatial and spatio-temporal data are two data types which are collected and used in research across a diverse array of domains.

Reviewer 4: ```suggestion
Although there is so much data being collected in multiple disciplines and made openly available, it can be difficult to find, retrieve and utilize these resources. Spatial and spatio-temporal data are two data types which are collected and used in research across a diverse array of domains.
```

---
title: The Story Behind rspatialdata
author:
  - Dilinie Seimon
  - Varsha Ujjinni Vijay Kumar
date: '2021-09-14'
slug: rspatialdata
categories: []
tags:

>   - spatial

Reviewer 4: ```suggestion
  - rnaturalearth
  - osmdata
  - MODIStsp
  - nasapower
  - rdhs
  - open-data
  - spatial
```

  - geospatial
  - tutorial
  - community

> description: Using rOpenSci packages to create cool maps

Reviewer 4: Definitely want to show case rspatialdata!

```suggestion
description: "rspatialdata: tutorials for working with spatial data using R, featuring many rOpenSci packages!"
```

> output:

Reviewer 4: If you decide to include the screen shot of the rspatialdata homepage, it'd be great to include as a twitter card image! You just need to update `name-of-image.png` in this suggestion (and when we get the proper publication date, the date)

```suggestion
twitterImg: blog/2021/09/14/rspatialdata/name-of-image.png
twitterAlt: "Screenshot of rspatialdata homepage featuring multicoloured tiles each outlining a different type of spatial data"
output:
```

  html_document:
    keep_md: yes
---

> As two postgrad students on summer vacation but no travel plans (during this global pandemic of course), we took up an internship at King Abdullah University of Science and Technology (KAUST) last summer and ended up collaborating on a really cool project with [Paula Moraga](https://www.paulamoraga.com/)! Did we mention that we worked in Saudi Arabia, while living in Australia? All is well when we have the internet.

Author: Thanks @steffilazerte! Just a bit of humour to get through these crazy times!

With the R programming language becoming increasingly popular among academics, researchers, and scientists, and spatial data being easier to interpret when visualized, Paula thought it was a great idea to create a repository of data sources and simple tutorials on how to retrieve and visualize spatial data using R.

This idea then became [rspatialdata](https://rspatialdata.github.io/) - a collection of data sources and tutorials on downloading and visualising spatial data using R.

rspatialdata consists of a non-exhaustive list of R-packages which have been developed as clients for different spatio-temporal databases and APIs. It also consists of tutorials on how to use these R packages, understand the different types of spatial data objects available and create visualizations using them. After doing a lot of research, we managed to pick out a few R-packages which we were confident to give out up-to-date data from reliable sources.

Coincidentally, most of what we picked out happened to be rOpenSci packages! We were amazed at how rich and easy to use most of these packages were. So here is our experience using those packages and some things we really enjoyed doing!

> ## Downloading boundaries of countries using [rnaturalearth](https://cran.r-project.org/web/packages/rnaturalearth/README.html)

Reviewer 4: Here and for all the other rOpenSci packages, please link to the rOpenSci docs:
```suggestion
## Downloading boundaries of countries using [rnaturalearth](https://docs.ropensci.org/rnaturalearth)
```

> The rnaturalearth package facilitates communication with [Natural earth](http://www.naturalearthdata.com/) map data. It allows us to easily download and visualise boundaries of countries and boundaries of states within countries as well.

Reviewer 4: ```suggestion
The rnaturalearth package (by [Andy South](https://github.com/andysouth)) facilitates communication with [Natural earth](http://www.naturalearthdata.com/) map data. It allows us to easily download and visualise boundaries of countries and boundaries of states within countries as well.
```

We used the osmdata package to download and visualize hospitals in Lagos, Nigeria. Then we also downloaded spatial data for different types of highways, streets and waterways in Lagos, and created the following map entirely using data retrieved through osmdata!

{{< figure src = "hospitals-lagos-nigeria.jpg" width = "600" alt = "Hospitals in Lagos" class = "center" link = "https://rspatialdata.github.io/osm.html">}}

Have a read through our [Open Street Map data tutorial](https://rspatialdata.github.io/osm.html) for a complete tutorial on how to install the osmdata package, find what spatial features are available, learn how to download these features and also create some cool maps (including an interactive one)!

## Visualizing vegetation in Mongolia and land cover in Zimbabwe using [MODIStsp](https://cran.r-project.org/web/packages/MODIStsp/index.html)

MODIS(Moderate Resolution Imaging Spectroradiometer) is an instrument aboard the NASA Terra and Aqua satellites, which orbits the entire Earth every 1-2 days, acquiring data at different spatial resolutions. The data acquired by MODIS describes features of the land, oceans and the atmosphere.

> MODIStsp - an acronym for ‘MODIS Time Series Processing’, is an R package for downloading and pre-processing time series raster data from MODIS [data products](https://modis.gsfc.nasa.gov/data/dataprod/).

Reviewer 4: ```suggestion
MODIStsp - an acronym for ‘MODIS Time Series Processing’ (by (Lorenzo Busetto)[https://github.com/lbusett] and [Luigi Ranghetti](https://github.com/ranghetti)), is an R package for downloading and pre-processing time series raster data from MODIS [data products](https://modis.gsfc.nasa.gov/data/dataprod/).
```
